src/components/clusters/PullSecret.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { Checkbox, FormGroup, Popover } from '@patternfly/react-core';
+import { ExternalLinkAltIcon, OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
+import { ClusterCreateParams, ocmClient } from '../../api';
+import { CLUSTER_MANAGER_SITE_LINK, PULL_SECRET_INFO_LINK } from '../../config';
+import { TextAreaField } from '../ui';
+import { useFormikContext } from 'formik';
+
+export type PullSecretProps = {
+  pullSecret?: string;
+};
+
+const ClusterManagerSiteLink = () => (
+  <a href={CLUSTER_MANAGER_SITE_LINK} target="_blank" rel="noopener noreferrer">
+    OpenShift Cluster Manager <ExternalLinkAltIcon />
+  </a>
+);
+
+const PullSecretInfoLink = () => (
+  <a href={PULL_SECRET_INFO_LINK} target="_blank" rel="noopener noreferrer">
+    Learn more about pull secrets <ExternalLinkAltIcon />.
+  </a>
+);
+
+const pullSecretHelperText = ocmClient ? (
+  <>
+    Your Red Hat account's pull secret is used by default.&nbsp;
+    <PullSecretInfoLink />
+  </>
+) : (
+  <>
+    A Red Hat account's pull secret can be found in &nbsp;
+    <ClusterManagerSiteLink />
+  </>
+);
+
+const PullSecretInfo = () => (
+  <Popover
+    position="top"
+    bodyContent={
+      <>
+        Pull secrets are used to download OpenShift Container Platform components and connect
+        clusters to a Red Hat account.
+        {ocmClient ? (
+          <PullSecretInfoLink />
+        ) : (
+          <>
+            Pull secrets can be found in&nbsp;
+            <ClusterManagerSiteLink />
+          </>
+        )}
+      </>
+    }
+  >
+    <OutlinedQuestionCircleIcon />
+  </Popover>
+);
+
+const PullSecret: React.FC<PullSecretProps> = ({ pullSecret }) => {
+  // Fetched pull secret will never change - see LoadingState in NewCluster
+  const [isExpanded, setExpanded] = React.useState(!pullSecret);
+  const { setFieldValue } = useFormikContext<ClusterCreateParams>();
+
+  const textArea = (
+    <TextAreaField
+      name="pullSecret"
+      label="Pull Secret"
+      labelIcon={ocmClient ? undefined : <PullSecretInfo />}
+      getErrorText={(error) => (
+        <>
+          {error} {pullSecretHelperText}
+        </>
+      )}
+      helperText={pullSecretHelperText}
+      isRequired
+    />
+  );
+
+  if (ocmClient) {
+    const onCheckboxChange = () => {
+      if (isExpanded) {
+        // about to collapse, reset to original value
+        setFieldValue('pullSecret', pullSecret);
+      }
+      setExpanded(!isExpanded);
+    };
+
+    return (
+      <FormGroup
+        fieldId="checkbox-pull-secret"
+        label={
+          <Checkbox
+            id="checkbox-pull-secret"
+            className="pf-u-display-inline-flex"
+            aria-label="edit pull secret"
+            isChecked={isExpanded}
+            onChange={onCheckboxChange}
+            label="Edit pull secret"
+          />
+        }
+        labelIcon={<PullSecretInfo />}
+      >
+        {isExpanded && textArea}
+      </FormGroup>
```

Comment:
```suggestion
      <>
        <Checkbox
          id="checkbox-pull-secret"
          className="pf-u-display-inline-flex"
          aria-label="edit pull secret"
          isChecked={isExpanded}
          onChange={onCheckboxChange}
          label={
            <>
              Edit pull secret <PullSecretInfo />
            </>
          }
        />
        {isExpanded && textArea}
      </>
```

Standalone checkbox seems to work fine unless I am missing something.

Reply:
The `FormGroup` was added in attempt to fix alignment and event propagation. It's a good idea to wrap the icon at lower level, so I am reverting this change here.